### ISSUE TITLE
ImageViewer's UI auto magnify with no sense?

### ISSUE THREAD

**Issue by JeremyWu917** [NaBian/HandyControl#1328]
### Describe the bug

 ImageViewer's UI auto update when new ImageSource been given but the ScaleStr not changed

#### MainWindow.xaml
```xaml
<hc:GlowWindow
    x:Class="WpfApp1.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:hc="https://handyorg.github.io/handycontrol"
    xmlns:local="clr-namespace:WpfApp1"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    Title="MainWindow"
    Width="600"
    Height="620"
    ActiveGlowColor="Red"
    WindowStartupLocation="CenterScreen"
    mc:Ignorable="d">
    <StackPanel Margin="1" Orientation="Vertical">
        <Button
            x:Name="ButtonChangeMode"
            Margin="10"
            Click="ButtonChangeMode_Click"
            Content="Do/Stop"
            Style="{StaticResource ButtonPrimary}" />
        <hc:ImageViewer
            x:Name="ImageViewerMain"
            Width="500"
            Height="500"
            Margin="10"
            Background="{DynamicResource SecondaryRegionBrush}"
            IsFullScreen="True"
            IsHitTestVisible="True"
            ShowImgMap="True"
            ShowToolBar="True" />
    </StackPanel>
</hc:GlowWindow>
```
#### MainWindow.xaml.cs
```C#
using HandyControl.Themes;
using System.Collections.Generic;
using System.IO;
using System.Threading;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Media.Imaging;
using System.Windows.Threading;

namespace WpfApp1
{
    /// <summary>
    /// Interaction logic for MainWindow.xaml
    /// </summary>
    public partial class MainWindow
    {
        public MainWindow()
        {
            InitializeComponent();
            ThemeManager.Current.ApplicationTheme = ApplicationTheme.Dark;
            List<string> list = new()
            {
                "R.bmp",
                "G.bmp",
                "B.bmp",
            };
            source = new();
            Task.Run(async () =>
            {
                while (!source.IsCancellationRequested)
                {
                    for (int i = 0; i < list.Count; i++)
                    {
                        await Dispatcher.BeginInvoke(() =>
                        {
                            ImageViewerMain.ImageSource = BitmapFrame.Create(new FileStream(list[i], FileMode.Open, FileAccess.Read, FileShare.Read));
                        });
                        await Task.Delay(3000);
                    }
                }
            }, source.Token);
        }
        CancellationTokenSource source = new();
        string tag = "run";
        private void ButtonChangeMode_Click(object sender, RoutedEventArgs e)
        {
            if ("run" != tag)
            {
                source = new();
                ImageViewerMain.Tag = "run";
            }
            else
            {
                source.Cancel();
                ImageViewerMain.Tag = "stop";
            }
        }
    }
}

```

#### Run the Application

> **Note**
> There are three pictures named R.bmp G.bmp and B.bmp in your exe path, and they have same width, height and size.

![1](https://user-images.githubusercontent.com/54832272/209077287-bec04251-ca15-49bd-a48a-dad6159d7edc.gif)

### Expected behavior

The ImageViewer's UI _DO Not_ magnify when same pic been given.

### NuGet package version

HandyControl 3.4.0

### IDE

Visual Studio 2022

### Framework type

.Net 6.0

### Windows version

Windows 11 (22000)

**Comment by NaBian** [comment]
fixed in b818e7a4

**Comment by JeremyWu917** [comment]
@NaBian This PR fired a new issues here #1373 , please help to check and advise, thanks
